src/PKSim.Infrastructure/Services/ImportObservedDataTask.cs

```diff
@@ -164,7 +164,10 @@ private void AddObservedDataFromConfiguration(ImporterConfiguration configuratio
 
       private DataRepository findDataRepositoryInList(IEnumerable<DataRepository> dataRepositoryList, DataRepository targetDataRepository)
       {
-         return (from dataRepo in dataRepositoryList let result = targetDataRepository.ExtendedProperties.KeyValues.All(keyValuePair => dataRepo.ExtendedProperties[keyValuePair.Key].ValueAsObject.ToString() == keyValuePair.Value.ValueAsObject.ToString()) where result select dataRepo).FirstOrDefault();
+         return (from dataRepo in dataRepositoryList let result = targetDataRepository.ExtendedProperties.KeyValues.All(keyValuePair => 
```

Comment:
That feels that some exact same code that we had in core. Why do we have it twice?

Reply:
or am I mistaken?